apps/01PigLatin.js

```diff
@@ -7,12 +7,55 @@ var rl = readline.createInterface({
   output: process.stdout
 });
 
+//after the user types in a word and hits the enter key, the code will start running.
 
 function pigLatin(word) {
 
-  // Your code here
+  word = word.toLowerCase();
+  var firstLetter = word[0];
 
-}
+  var vowelIndex = -1; // Set it to assume there are no vowels
+
+  if ( ( word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex ) || vowelIndex === -1 ) {
+    vowelIndex = word.indexOf('a');
+  }
+
+  if ( ( word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex ) || vowelIndex === -1 ) {
+    vowelIndex = word.indexOf('e');
+  }
+
+  if ( ( word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex ) || vowelIndex === -1 ) {
+    vowelIndex = word.indexOf('i');
+  }
+
+  if ( ( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex ) || vowelIndex === -1 ) {
+    vowelIndex = word.indexOf('o');
+  }
+
+  if ( ( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex ) || vowelIndex === -1 ) {
+    vowelIndex = word.indexOf('u');
+  }
+
+  if ( ( word.indexOf('y') > -1 && word.indexOf('y') < vowelIndex ) || vowelIndex === -1 ) {
+    vowelIndex = word.indexOf('y');
+  }
+
+  
+  
+  var firstPart = word.slice(0, vowelIndex);
+  var restWord  = word.slice(vowelIndex, word.length)
+
+//Conditional added to notify the user if anything other than a word is input.
+
+  if(vowelIndex > 0) {
+    return restWord + firstPart + 'ay';
+  } else if (vowelIndex === 0) {
```

Comment:
This solution for returning words beginning with vowels is great and much more simple than the way I did it. Great work.